mithril-common/src/cardano_transaction_parser.rs

```diff
@@ -127,10 +134,26 @@ impl CardanoTransactionParser {
                     immutable_file.path
                 )
             })?;
-            blocks.push(CardanoTransactionParser::convert_to_block(
-                &block,
-                immutable_file,
-            )?);
+            match CardanoTransactionParser::convert_to_block(&block, immutable_file) {
+                Ok(convert_to_block) => {
+                    blocks.push(convert_to_block);
+                }
+                // TODO: no error are returned due to the crate 'pallas-hardano'
+                // that doesn't support all encoding versions (test networks doesn't work)
+                Err(e) if self.allow_unparsable_block => {
+                    error!(
+                        self.logger,
+                        "pallas-hardano does not support this block format"
+                    );
+                    error!(self.logger, "error='{e}'.");
+                    let mut potential_source = e.source();
+                    while let Some(error) = potential_source {
+                        error!(self.logger, "cause='{}'.", error);
+                        potential_source = error.source();
+                    }
```

Comment:
This could probably do the job:
```suggestion
                    error!(
                        self.logger,
                        "the cbor encoded block could not be parsed";
                        "error" => ?err, "immutable_file" => immutable_file
                    );
```

Reply:
Is it ok cause it prints the full backtrace (which is quite verbose):
```
Mar 13 16:53:09.977 ERRO the cbor encoded block could not be parsed, immutable_file_number: 4831, error: Error while decoding block in immutable file: '"../mithril-test-lab/test_data/immutable/04831.chunk"'

Caused by:
    Invalid CBOR structure: decode error: unknown data type of maybe indef array

Stack backtrace:
   0: std::backtrace_rs::backtrace::libunwind::trace
             at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/../../backtrace/src/backtrace/libunwind.rs:104:5
   1: std::backtrace_rs::backtrace::trace_unsynchronized
             at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2: std::backtrace::Backtrace::create
             at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/backtrace.rs:331:13
   ...
   ...
   ...
  49: std::thread::Builder::spawn_unchecked_::{{closure}}
             at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/thread/mod.rs:528:30
  50: core::ops::function::FnOnce::call_once{{vtable.shim}}
             at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/core/src/ops/function.rs:250:5
  51: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/alloc/src/boxed.rs:2015:9
  52: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/alloc/src/boxed.rs:2015:9
  53: std::sys::unix::thread::Thread::new::thread_start
             at /rustc/07dca489ac2d933c78d3c5158e3f43beefeb02ce/library/std/src/sys/unix/thread.rs:108:17
  54: __pthread_joiner_wake
```

Reply:
If it's too long, `"error" => err` should produce succinct error message